src/test/java/org/elasticsearch/index/query/TemplateQueryTest.java

```diff
@@ -371,6 +376,45 @@ public void testIndexedTemplate() throws Exception {
         assertHitCount(sr, 4);
     }
 
+    // Relates to #10397
+    @Test
+    public void testIndexedTemplateOverwrite() throws Exception {
+        createIndex("testindex");
+        ensureGreen("testindex");
+
+        index("testindex", "test", "1", jsonBuilder().startObject().field("searchtext", "dev1").endObject());
+        refresh();
+
+        PutIndexedScriptResponse scriptResponse = client().preparePutIndexedScript(MustacheScriptEngineService.NAME, "git01", 
+                "{\"query\": {\"match\": {\"searchtext\": {\"query\": \"{{P_Keyword1}}\",\"type\": \"ooophrase_prefix\"}}}}").get();
+        assertTrue(scriptResponse.isCreated());
+
+        GetIndexedScriptResponse getResponse = client().prepareGetIndexedScript(MustacheScriptEngineService.NAME, "git01").get();
+        assertTrue(getResponse.isExists());
+
+        Map<String, Object> templateParams = Maps.newHashMap();
+        templateParams.put("P_Keyword1", "dev");
+
+        try {
+            SearchResponse searchResponse = client().prepareSearch("testindex").setTypes("test").
+                setTemplateName("git01").setTemplateType(ScriptService.ScriptType.INDEXED).setTemplateParams(templateParams).get();
+            fail("Broken test template is parsing w/o error.");
+        } catch (SearchPhaseExecutionException e) {
+            // the above is expected to fail
+        }
+
+        PutIndexedScriptRequestBuilder builder = client()
+                .preparePutIndexedScript(MustacheScriptEngineService.NAME, "git01",
+                        "{\"query\": {\"match\": {\"searchtext\": {\"query\": \"{{P_Keyword1}}\",\"type\": \"phrase_prefix\"}}}}")
+                .setOpType(OpType.INDEX);
+        scriptResponse = builder.get();
+        assertEquals(scriptResponse.getVersion(), 2);
+        SearchResponse searchResponse = client().prepareSearch("testindex").setTypes("test").
+                setTemplateName("git01").setTemplateType(ScriptService.ScriptType.INDEXED).setTemplateParams(templateParams).get();
+        assertHitCount(searchResponse, 1);
```

Comment:
I would love to have some more coverage here, maybe update a few more times (`randomIntBetween(1,10)` ?) and verify that updates always make it through, not just once?

Also, can we write a test for plain indexed scripts too (rather than search templates)?